compiler/rustc_target/src/asm/mod.rs

```diff
@@ -312,6 +312,7 @@ impl InlineAsmReg {
             InlineAsmArch::RiscV32 | InlineAsmArch::RiscV64 => {
                 Self::RiscV(RiscVInlineAsmReg::parse(name)?)
             }
+            #[allow(unreachable_code)] // NvptxInlineAsmReg::parse does not return.
             InlineAsmArch::Nvptx64 => Self::Nvptx(NvptxInlineAsmReg::parse(name)?),
```

Comment:
feels like here the issue is that `NvptxInlineAsmReg` should only be privately uninhabited?

Reply:
was the stabilization of `feature(min_exhaustive_patterns)` enough to unblock this PR?